bugfixtemplates/getting-substring-before-suffix/GettingSubstringBeforeSuffix.java

```diff
@@ -0,0 +1,135 @@
+package bugfixtemplates;
+
+class GettingSubstringBeforeSuffix {
+
+    Object mutex, mutex1;
+
+    public void m1(String className) {
+        int i = className.lastIndexOf(SUFFIX);
+        className.substring(0, i);
+    }
+
+    public void m2(String className) {
+        int i = className.lastIndexOf(SUFFIX);
+        String subclassPrefix = i > 0 ? className.substring(0, i) : className;
+    }
+
+    public void m3(String className) {
+        int i = className.indexOf(SUFFIX);
+        className.substring(0, i);
+    }
+
+    public void m4(String className) {
+        int i = className.indexOf(SUFFIX);
+        String subclassPrefix = i > 0 ? className.substring(0, i) : className;
+    }
+
+    // https://guru-reviews-beta.corp.amazon.com/feedback/internal/CR-11623479/1/
+    public String getAttributeKey(String aggregationName) {
+        if (aggregationName == null) {
+            return null;
+        }
+        if (aggregationName.endsWith(INTEGER_AGG)) {
+            return aggregationName.substring(0, aggregationName.lastIndexOf(INTEGER_AGG));
+        } else if (aggregationName.endsWith(LONG_AGG)) {
+            return aggregationName.substring(0, aggregationName.lastIndexOf(LONG_AGG));
+        } else if (aggregationName.endsWith(STRING_AGG)) {
+            return aggregationName.substring(0, aggregationName.lastIndexOf(STRING_AGG));
+        }
+        return null;
+    }
+
+    public void m5(String className) {
+        int i = className.indexOf(SUFFIX);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 43 does not find the substring and returns -1, the call to substring() on line 44 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string className.